src/server/replica.cc

```diff
@@ -86,19 +86,18 @@ Replica::~Replica() {
 
 static const char kConnErr[] = "could not connect to master: ";
 
-error_code Replica::Start(facade::SinkReplyBuilder* builder) {
+GenericError Replica::Start() {
   VLOG(1) << "Starting replication";
   ProactorBase* mythread = ProactorBase::me();
   CHECK(mythread);
 
-  auto check_connection_error = [this, builder](error_code ec, const char* msg) -> error_code {
+  auto check_connection_error = [this](error_code ec, const char* msg) -> GenericError {
```

Comment:
can RETURN_ON_ERR work on GenericError?

Reply:
Generic error is implicitly convertible to error code. However RETURN_ON_ERR does this:

`GenericError -> ErrorCode` then RETURN_ON_ERR returns that ErrorCode, which is then converted back to GenericError. 

This looses the `details` part of the GenericError. I will fix this. 

I am SAVING my rant around this.